shared/js/blobview.js

```diff
@@ -457,26 +373,15 @@ var BlobView = (function() {
       if (size % 2) {
         fail('size must be a multiple of two');
       }
-      if (le === null || le === undefined) {
-        var BOM = this.readUnsignedShort();
-        size -= 2;
-        if (BOM === 0xFEFF) {
-          le = false;
-        } else {
-          le = true;
-        }
-      }
 
-      var s = '';
-      for (var i = 0; i < size; i += 2) {
-        var charcode = this.getUint16(this.index + i, le);
-        if (charcode === 0) {
-          i += 2;
+      for (var len = 0; len < size; len += 2) {
+        if (this.getUint16(this.index + len, le) === 0) {
           break;
         }
-        s += String.fromCharCode(charcode);
       }
-      this.index += i;
+
+      var s = this.getUTF16Text(this.index, len, le);
```

Comment:
Yeah, the others are like that too (they have an `indexOf` call), but I think this is still fast in practice. It's also only called for older versions of ID3 tags.